internal/diag.go

```diff
@@ -232,6 +264,29 @@ LOOP:
 	return nil
 }
 
+func getOperatorLabel(kc *Kubectl, ns string) (labels.Set, error) {
+	saList, err := kc.Clientset.CoreV1().ServiceAccounts(ns).List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("while retrieving list of serviceaccounts in ns '%s': %w", ns, err)
+	}
+	if saList == nil {
+		return nil, fmt.Errorf("nil sa while retrieving list of serviceaccounts in ns '%s'", ns)
+	}
```

Comment:
This is not really necessary as we can iterate over a nil slice and then we will return `fmt.Errorf("unable to find serviceaccount corresponding to eck operator")` which has a slightly more informative message IMO.

```suggestion
```